components/proxy/src/main/java/com/hotels/styx/routing/RoutingObjectAdapter.java

```diff
@@ -0,0 +1,67 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.routing;
+
+import com.hotels.styx.api.Eventual;
+import com.hotels.styx.api.HttpInterceptor;
+import com.hotels.styx.api.LiveHttpRequest;
+import com.hotels.styx.api.LiveHttpResponse;
+import com.hotels.styx.api.ResponseEventListener;
+import com.hotels.styx.api.extension.loadbalancing.spi.LoadBalancingMetric;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.LongAdder;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Adapts routing object into Styx core.
+ * Provides a load balancing metric for the adapted routing object.
+ */
+public class RoutingObjectAdapter implements RoutingObject {
```

Comment:
This is called "Adapter" but the interface is the same as the delegate's interface. It seems that what it actually does is not adapting interfaces, but rather wrapping to add metrics.

Perhaps `RequestCountingRoutingObject`, `InstrumentedRoutingObject` or something like that might be a better name?

Reply:
See my response to @dvlato above.

It "adapts" a bare RoutingObject into styx core. It adds any functionality or metadata that may be necessary for styx to interoperate and use the routing object.

Let's add additional comments to the other, above thread.